python-lib/onedrive_client.py

```diff
@@ -128,10 +131,21 @@ def get_item(self, path):
         return onedrive_item
 
     def get_shared_directory_drive_id(self, shared_directory_name):
-        url = "https://graph.microsoft.com/v1.0/me/drive/root:/{}:".format(shared_directory_name)
-        item = self.get(url, headers=self.generate_header())
-        drive_id = get_value_from_path(item, ["remoteItem", "parentReference", "driveId"])
-        return drive_id
+        if not shared_directory_name:
+            return None
+        drive_name = None
+        if shared_directory_name.startswith("https://"):
+            base_64_sharing_url = base64.b64encode(shared_directory_name.encode("utf-8")).decode("utf-8")
+            base_64_sharing_url = base_64_sharing_url.replace("/", "_").replace("+", "-")
```

Comment:
from the doc https://learn.microsoft.com/en-us/graph/api/shares-get?view=graph-rest-1.0&tabs=http#:~:text=encode%20the%20URL.-,Convert,-the%20base64%20encoded
Convert the base64 encoded result to [unpadded base64url format](https://en.wikipedia.org/wiki/Base64) **by removing = characters from the end of the value**, replacing / with _ and + with -.)

```suggestion
            base_64_sharing_url = base_64_sharing_url.rstrip("=").replace("/", "_").replace("+", "-")
```